internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -363,6 +412,128 @@ func (cf CloudFormation) executeAndRenderChangeSet(in *executeAndRenderChangeSet
 	return nil
 }
 
+func (cf CloudFormation) waitForSignalAndHandleInterrupt(ctx context.Context, cancelFn context.CancelFunc, sigCh chan os.Signal, stackName string) error {
+	for {
+		select {
+		case <-sigCh:
+			cancelFn()
+			stopCatchSignals(sigCh)
+			stackDescr, err := cf.cfnClient.Describe(stackName)
+			if err != nil {
+				return fmt.Errorf("describe stack %s: %w", stackName, err)
+			}
+			switch aws.StringValue(stackDescr.StackStatus) {
+			case sdkcloudformation.StackStatusCreateInProgress:
+				log.Infof(`Received Interrupt for Ctrl-C.
+Pressing Ctrl-C again will exit immediately but the deletion of stack %s will continue
+`, stackName)
+				description := fmt.Sprintf("Delete stack %s", stackName)
+				if err := cf.deleteAndRenderStack(stackName, description, func() error {
+					return cf.cfnClient.DeleteAndWait(stackName)
+				}); err != nil {
+					return err
+				}
+				return &ErrStackDeletedOnInterrupt{stackName: stackName}
+			case sdkcloudformation.StackStatusUpdateInProgress:
+				log.Infof(`Received Interrupt for Ctrl-C.
+Pressing Ctrl-C again will exit immediately but stack %s rollback will continue
+`, stackName)
+				description := fmt.Sprintf("Canceling stack update %s", stackName)
+				if err := cf.cancelUpdateAndRender(&cancelUpdateAndRenderInput{
+					stackName:   stackName,
+					description: description,
+					cancelUpdateFn: func() error {
+						return cf.cfnClient.CancelUpdateStack(stackName)
+					},
+				}); err != nil {
+					return err
+				}
+				return &ErrStackUpdateCanceledOnInterrupt{stackName: stackName}
+			}
+			return nil
+		case <-ctx.Done():
+			stopCatchSignals(sigCh)
+			return nil
+		}
+	}
+}
+
+type cancelUpdateAndRenderInput struct {
+	stackName      string
+	description    string
+	cancelUpdateFn func() error
+}
+
+func (cf CloudFormation) cancelUpdateAndRender(in *cancelUpdateAndRenderInput) error {
+	stackDescr, err := cf.cfnClient.Describe(in.stackName)
+	if err != nil {
+		return fmt.Errorf("describe stack %s: %w", in.stackName, err)
+	}
+	if stackDescr.ChangeSetId == nil {
+		return fmt.Errorf("ChangeSetID not found for stack %s", in.stackName)
+
+	}
+	waitCtx, cancelWait := context.WithTimeout(context.Background(), waitForStackTimeout)
+	defer cancelWait()
+	g, ctx := errgroup.WithContext(waitCtx)
```

Comment:
since we don't use the wait ctx any differently, let's name them the same to avoid unnecessary variables
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), waitForStackTimeout)
	defer cancel()
	g, ctx := errgroup.WithContext(ctx)
```